numba_dpex/core/codegen.py

```diff
@@ -55,44 +57,22 @@ def _optimize_final_module(self):
         # Make optimization level depending on config.DPEX_OPT variable
         pmb.opt_level = config.DPEX_OPT
         if config.DPEX_OPT > 2:
-            logging.warning(
+            warnings.warn(
                 "Setting NUMBA_DPEX_OPT greater than 2 known to cause issues "
                 + "related to very aggressive optimizations that leads to "
                 + "broken code."
             )
 
         pmb.disable_unit_at_a_time = False
 
-        if config.INLINE_THRESHOLD is not None:
-            # Check if a decorator-level inline threshold was set and use that
-            # instead of the global configuration.
-            if (
-                hasattr(self, "_inline_threshold")
-                and self._inline_threshold > 0
-                and self._inline_threshold <= 3
-            ):
-                logging.warning(
-                    "Setting INLINE_THRESHOLD leads to very aggressive "
-                    + "optimizations that may produce incorrect binary."
-                )
-                pmb.inlining_threshold = self._inline_threshold
-            elif not hasattr(self, "_inline_threshold"):
-                logging.warning(
-                    "Setting INLINE_THRESHOLD leads to very aggressive "
-                    + "optimizations that may produce incorrect binary."
-                )
-                pmb.inlining_threshold = config.INLINE_THRESHOLD
-        else:
-            if (
-                hasattr(self, "_inline_threshold")
-                and self._inline_threshold > 0
-                and self._inline_threshold <= 3
-            ):
-                logging.warning(
-                    "Setting INLINE_THRESHOLD leads to very aggressive "
-                    + "optimizations that may produce incorrect binary."
+        if hasattr(self, "_inline_threshold") and self._inline_threshold > 0:
```

Comment:
```suggestion
        pmb.inlining_threshold = self.inline_threshold
```
why not to set it unconditionally? We can warn in the `@inline_threshold.setter`

Reply:
Makes sense. Done